backend/api/v1/schools.py

```
@@ -1,31 +1,117 @@
+    try:
+        school_id = current_user["school_id"]
+        response = supabase_client.table('schools').select("*").eq('id', str(school_id)).single().execute()
+        
+        if response.data:
+            return response.data
+        
+        raise HTTPException(status_code=404, detail="School not found")
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Error fetching school: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)